velodyne_pointcloud/src/lib/rawdata.cc

```diff
@@ -458,14 +494,21 @@ namespace velodyne_rawdata
             intensity = (intensity > max_intensity) ? max_intensity : intensity;
 
             if (pointInRange(distance)) {
+              // Set point time as beginning of scan and then apply timing offset:
+              ros::Time pt_time = pkt.stamp;
+              if (!timing_offsets_.empty()) {
+                  pt_time = pt_time + ros::Duration(timing_offsets_[block][firing_seq]);
```

Comment:
what's the relationship between this time_offset patching and the one that happens in `getPacket()`?  Following this code, I initially assumed `timing_offsets_` values would get passed to `getPacket()`..

Reply:
I think I'm missing the high level picture here.

It looks like all packets in a scan used to get timestamps with .. something (average between the scan start and the process time of each packet? or maybe the end of the scan?).

Now, we're using the start time of the scan... plus the computed time offset?

Can you fill in the gaps for me?

Reply:
Added comments to explain.